### ISSUE TITLE
object detection and image orientation

### ISSUE THREAD

**Issue by romanoss** [apple/turicreate#1675]
Hi,
I am training the yolo net with image data, where I use portrait training images for some classes and landscape images for other classes. When I test the model against landscape images I get significantly better results for classes that were trained with landscape images.
Is that reasonable, is that how it works or do I miss something?

kind regards
Roman

**Comment by nickjong** [comment]
It seems theoretically possible for the aspect ratio to matter somewhat, but in practice our data augmentation pipeline randomly samples cropped versions of the raw images, and aspect ratio of the cropped version is mostly independent of the original aspect ratio (subject just to some constraints on fit).

Is it possible this has more to do with the classes that happen to have landscape training examples versus the ones that happen to have portrait training examples?

Are you measuring accuracy using a test/validation set, or is this anecdotal based on what is deployed to Core ML?

**Comment by romanoss** [comment]
Thx for the reply!

>Is it possible this has more to do with the classes that happen to have landscape training examples >versus the ones that happen to have portrait training examples?
I don't think - I am training images of paintings from a gallery

>Are you measuring accuracy using a test/validation set, or is this anecdotal based on what is deployed >to Core ML?
training data are close up images of the paintings and test images are from images shot from a greater distance - like when you walk by. Measured against the trained model inside turicreate.

Let me do some more tests.

**Comment by nickjong** [comment]
Hmm, ideally there isn't a systematic difference between the training and test images. What happens if you combine them and then randomly split them? Or do you have just one (close-up) training image per class?

**Comment by romanoss** [comment]
I think I have an idea what's happening.
The breakfast sample uses a 640x480 detection view and stretches it to the device bounds.
I think when one feeds an image to the coreml model this image is distorted when using the camera in landscape mode or distorted and rotated if you use the camera in portrait mode. Therefore one gets better results while using the app in landscape mode (the difference is big)(maybe the banana class doesn't show this, because of the symmetries of the sample class). The solution of #1016 helps but is not optimal.
If I am right, you should do a little rework on the sample app.
Hmm?

**Comment by nickjong** [comment]
Ah, that makes sense. Just to follow up here, I'm not sure about the confidence values you're seeing, but ultimately we're tracking that problem with #2093
